### PR TITLE
refactor: migrate git revert to Git::Commands::Revert::*

### PR DESCRIPTION
**Pull request by jcouball**
Review our [guidelines for contributing](https://github.com/ruby-git/ruby-git/blob/main/CONTRIBUTING.md) to this repository. A good start is to:

- Write tests for your changes
- Run `rake` before pushing
- Include / update docs in the README.md and in YARD documentation

# Description

Migrates `git revert` from the monolithic `Git::Lib` to dedicated
`Git::Commands::Revert::*` classes as part of the ongoing Strangler Fig
architectural redesign (phase 2, item 41/54).

## Changes

**New command classes** (`lib/git/commands/revert/`):

| Class | Git command |
|---|---|
| `Git::Commands::Revert::Start` | `git revert <commit>...` |
| `Git::Commands::Revert::Continue` | `git revert --continue` |
| `Git::Commands::Revert::Skip` | `git revert --skip` |
| `Git::Commands::Revert::Abort` | `git revert --abort` |
| `Git::Commands::Revert::Quit` | `git revert --quit` |

`Revert::Start` supports the full option set: `--[no-]edit` / `-e`,
`--cleanup`, `--no-commit` / `-n`, `--gpg-sign` / `-S`, `--[no-]signoff` / `-s`,
`--strategy`, `--strategy-option` / `-X`, `--[no-]rerere-autoupdate`, and
`--mainline` / `-m`.

`Revert::Continue` supports `--[no-]edit` for controlling the commit message editor.

**`Git::Lib#revert` updated** to delegate to `Revert::Start`. The public interface
is preserved.

## Test coverage

- **30** unit examples covering every option, alias, and edge case for all 5 classes
- **10** integration examples validating real git interactions, including conflict
  handling for sequencer commands (continue/skip/abort) and the no-op behavior of
  `--quit` when no session is active

## Checklist

- [x] I reviewed and applied the project's [AI Policy](../AI_POLICY.md). I understand
  and verify any AI-assisted changes included in this PR and ensured they meet
  quality, security, and licensing standards.
- [x] Tests added/updated as needed and `rake` passes locally.
- [x] Documentation updated where applicable (README and/or YARD)

### CODE REVIEW

File: lib/git/commands/revert/start.rb [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+      #
+      class Start < Git::Commands::Base
+        arguments do
+          literal 'revert'
+
+          # Edit commit message (SYNOPSIS: [--[no-]edit])
+          flag_option %i[edit e], negatable: true
+
+          # Commit behavior (SYNOPSIS: [-n])
+          flag_option %i[no_commit n]
```
[the remark this answers is not in the document]
Comment (jcouball) [reply lib/git/commands/revert/start.rb:41]: The PR description should remove the non-interactive execution model comment. Commands are supposed to implement the command faithfully.

File: lib/git/commands/revert/start.rb [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+          value_option %i[strategy_option X], inline: true, repeatable: true
+
+          # Conflict resolution
+          flag_option :rerere_autoupdate, negatable: true
+
+          end_of_options
+
+          # Commits to revert
+          operand :commit, repeatable: true, required: true
+        end
```
[the remark this answers is not in the document]
Comment (jcouball) [reply lib/git/commands/revert/start.rb:66]: Remove the --reference comment from the PR.

File: spec/unit/git/commands/revert/start_spec.rb [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+
+      it 'adds --no-edit when false' do
+        expect_command_capturing('revert', '--no-edit', '--', 'abc123').and_return(command_result(''))
+        command.call('abc123', edit: false)
+      end
+
+      it 'supports the :e alias' do
+        expect_command_capturing('revert', '--edit', '--', 'abc123').and_return(command_result(''))
+        command.call('abc123', e: true)
+      end
```
[the remark this answers is not in the document]
Comment (jcouball) [reply spec/unit/git/commands/revert/start_spec.rb:45]: Fix the PR.

File: lib/git/commands/revert/quit.rb [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        # @!method call(*, **)
+        #
+        #   @overload call()
+        #
+        #     Clear the sequencer state, leaving the working tree as-is
+        #
+        #     @return [Git::CommandLineResult] the result of calling
+        #       `git revert --quit`
+        #
+        #     @raise [Git::FailedError] if no revert is in progress
```
[the remark this answers is not in the document]
Comment (jcouball) [reply lib/git/commands/revert/quit.rb]: It looks like `git revert --quit` with no revert in progress simply returns exit status 0 with no output. Please confirm. If that is true, the update the YARD doc accordingly.

File: lib/git/lib.rb [hunk 5]
```diff
@@ -1304,18 +1305,13 @@ def clean(opts = {})
...
-      # Forcing --no-edit as default since it's not an interactive session.
+      assert_valid_opts(opts, REVERT_ALLOWED_OPTS)
       opts = { no_edit: true }.merge(opts)
-
-      args = build_args(opts, REVERT_OPTION_MAP)
-      args << commitish
-
-      command_capturing('revert', *args)
+      opts[:edit] = false if opts.delete(:no_edit)
+      Git::Commands::Revert::Start.new(self).call(commitish, **opts).stdout
```
[the remark this answers is not in the document]
Comment (jcouball) [reply lib/git/lib.rb:1314]: Removed the deprecation statement from the PR description.
